src/ppom-controller.ts

```diff
@@ -507,9 +540,17 @@ export class PPOMController extends BaseControllerV2<
   /*
    * The function will return true if data is not already fetched for current chain.
    */
-  #isDataRequiredForCurrentChain(): boolean {
+  #isDataRequiredForChain(chainId: string): boolean {
     const { chainStatus } = this.state;
-    return !chainStatus[this.#chainId]?.dataFetched;
+    return !chainStatus[chainId]?.dataFetched;
+  }
+
+  /*
+   * The function will return true if data is not already fetched for all chains.
+   */
+  #isDataRequiredForAllChains(): boolean {
+    const isDataRequiredForChain = this.#isDataRequiredForChain.bind(this);
+    return Object.keys(this.state.chainStatus).some(isDataRequiredForChain);
```

Comment:
[nit] the name of this method implies to me that data is missing for all the chains, but seems to test whether data is missing for some of the chains.

Reply:
renamed to `#isDataRequiredForAnyChain` https://github.com/MetaMask/ppom-validator/pull/85/commits/95ce0db5e1d4697857ce4bf50c93a966f0f59ad7